### PR TITLE
feat:[SSCA-1191]: Add ticketInfo to remediationApis.

### PR DESCRIPTION
**Pull request by shash-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57407)

### CODE REVIEW

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger ut6

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/ssca/ticket/TicketServiceRestClient.java [hunk 1]
```diff
@@ -29,4 +30,9 @@ Call<TicketResponseDto> createTicket(@Header(NextGenAuthenticationFilter.AUTHORI
   Call<List<TicketResponseDto>> getTickets(@Header(NextGenAuthenticationFilter.AUTHORIZATION_HEADER) String authToken,
       @Query("module") String module, @Query("identifiers") Map<String, List<String>> identifiers,
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/ticket/TicketServiceRestClient.java:31]: the api accepts Map<String,String> identifiers

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 2]
```diff
@@ -457,7 +480,32 @@ public Page<RemediationArtifactListingResponse> listRemediationArtifacts(String
             .map(artifactInfo -> RemediationTrackerMapper.mapArtifactInfoToArtifactListingResponse(artifactInfo, body))
             .filter(Objects::nonNull)
             .collect(Collectors.toList());
-    // TODO add ticket details
+
+    String authToken = API_KEY + ticketServiceUtils.getTicketServiceToken(accountId);
+    Map<String, List<String>> identifiers = new HashMap<>();
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:485]: this won't work as discussed

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 3]
```diff
@@ -457,7 +480,32 @@ public Page<RemediationArtifactListingResponse> listRemediationArtifacts(String
...
+        ticketServiceRestClientService.getTickets(authToken, "SSCA", identifiers, accountId, orgId, projectId);
+
+    Map<String, TicketResponseDto> ticketIdToDTO = ticketResponseDtos.stream().collect(
+        Collectors.toMap(TicketResponseDto::getId, ticketResponseDto -> ticketResponseDto));
+
+    Map<String, RemediationArtifactListingResponse> artifactIdToResponse = artifactListingResponses.stream().collect(
+        Collectors.toMap(RemediationArtifactListingResponse::getId, response -> response));
+
+    for (ArtifactInfo artifactInfo : artifactInfos) {
+      String ticketId = artifactInfo.getTicketId();
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:497]: add null check

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 4]
```diff
@@ -186,7 +187,13 @@ public RemediationDetailsResponse getRemediationDetails(
       response.setCreatedByName(userInfoOptional.get().getName());
       response.setCreatedByEmail(userInfoOptional.get().getEmail());
     }
-    // TODO fetch info for ticketId
+    if (remediationTracker.getTicketId() != null) {
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:190]: this is dupe code as well

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 5]
```diff
@@ -457,7 +468,29 @@ public Page<RemediationArtifactListingResponse> listRemediationArtifacts(String
             .map(artifactInfo -> RemediationTrackerMapper.mapArtifactInfoToArtifactListingResponse(artifactInfo, body))
             .filter(Objects::nonNull)
             .collect(Collectors.toList());
-    // TODO add ticket details
+
+    Map<String, String> identifiers = new HashMap<>();
+    identifiers.put("remediationId", remediationTrackerId);
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:473]: nit: keep this as a constant

File: ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java [hunk 6]
```diff
@@ -457,7 +468,29 @@ public Page<RemediationArtifactListingResponse> listRemediationArtifacts(String
...
+        Collectors.toMap(TicketResponseDto::getId, ticketResponseDto -> ticketResponseDto));
+
+    Map<String, RemediationArtifactListingResponse> artifactIdToResponse = artifactListingResponses.stream().collect(
+        Collectors.toMap(RemediationArtifactListingResponse::getId, response -> response));
+
+    for (ArtifactInfo artifactInfo : artifactInfos) {
+      String ticketId = artifactInfo.getTicketId();
+      String artifactId = artifactInfo.getArtifactId();
+      if (ticketId != null) {
+        TicketResponseDto ticketResponseDto = ticketIdToDTO.getOrDefault(ticketId, null);
```
Comment (Varsha-lalwani) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/remediation_tracker/RemediationTrackerServiceImpl.java:487]: don't do this separately, just pass the map to the above mapper in line 468

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]

**Review by Varsha-lalwani — APPROVED** [review]
Comments will be addressed in next PR

**Pull request merged**